### PR TITLE
LPD-38269

### PR DESCRIPTION
**Pull request by NemethNorbert**
[LPD-38269](https://liferay.atlassian.net/browse/LPD-38269)

The issue was caused by an infinite loop of trying to update the current locale to the default language. I solved it by updating the locale in the language selector with journal:updateSelectedLanguage when switching the defaultLanguage

I also noticed we are rendering the Translation Option and Translation Filter component when editing the default values, This is not the intended behaviour so I added them in an if block.

Please review the changes

### CODE REVIEW

**Comment by ambrinchaudhary** [comment]
This is a bug and should have a test, isn't it? Or is there any current test that covers it?

**Comment by jkappler** [comment]
This was found by manual testing @ambrinchaudhary, so afik, there are not tests for this, so we should add one

**Comment by NemethNorbert** [comment]
Hey @ambrinchaudhary , @jkappler 

I started to work on a playwright test for changing the default Language, will update the PR when its done

**Comment by NemethNorbert** [comment]
Hey @ambrinchaudhary , @jkappler 

I updated the PR with the test.

Thank you

**Comment by boton** [comment]
ci:test:content-management-playwright

File: modules/test/playwright/tests/journal-web/journalArticleAutosave.spec.ts [hunk 1]
```diff
@@ -95,6 +98,88 @@ autoSaveTest(
...
+
+		await journalEditStructureDefaultValuesPage.goto({
+			siteUrl: site.friendlyUrlPath,
+			structureName,
+		});
+
+		const defaultLanguageButton = page
+			.locator(
+				'#_com_liferay_journal_web_portlet_JournalPortlet_-change-default-language'
+			)
```
Comment (boton) [inline modules/test/playwright/tests/journal-web/journalArticleAutosave.spec.ts:164]: Can we use a more semantic locator? 

Like: 

```js
const defaultLanguageButton = page.getByRole('group', { name: 'Basic Information' })
    .getByRole('button', {name: 'Change'});
```

**Review by boton — CHANGES_REQUESTED** [review]
LGTM, but the test is always failing in my local.

image

@NemethNorbert is it working on your local?

I left some small improvements but the fail is related with `clickAndExpectToBeVisible` but I don't get the cause :(

image

Let's see if it at least works in the CI 😕

**Comment by boton** [comment]
Forget about it I finally get it, after clicking on `getByRole('button', {name: 'Change'})` we are in an infinite loop.

image 

@NemethNorbert can you recheck?

**Comment by NemethNorbert** [comment]
> Forget about it I finally get it, after clicking on `page.getByRole('menuitem', {name: 'ca_ES'})` we are in an infinite loop.
> 
> image
> 
> @NemethNorbert can you recheck?

Hey @boton ,

I think the issue is that you forgot to deploy the changes. The infinite loop is the issue I fixed. The test is passing locally.
I go ahead and implement the simplification on the test you requested and push it up.

Thanks

**Comment by boton** [comment]
> Hey @boton ,
> 
> I think the issue is that you forgot to deploy the changes. The infinite loop is the issue I fixed. The test is passing locally.
> I go ahead and implement the simplification on the test you requested and push it up.
> 
> Screenshot 2024-10-07 at 18 14 10
> 
> 
> Thanks

I'm sorry about it, I'm not on the pc but looks like I deployed badly, 

I try deploy it with a command but looks I do something bad, your explanation make totally sense.

Feel free to forward if it test pass on CI :)

**Comment by NemethNorbert** [comment]
ci:forward

**Comment by DiegoHu97** [comment]
ci:report:34384024

**Comment by DiegoHu97** [comment]
ci:test:content-management-playwright

**Comment by DiegoHu97** [comment]
Test passed in CI, forwarding the PR

**Comment by DiegoHu97** [comment]
ci:forward

**Comment by DiegoHu97** [comment]
Hey @NemethNorbert there is a SF issue, could you take a look? Thanks!

File: modules/test/playwright/tests/journal-web/journalArticleAutosave.spec.ts [hunk 2]
```diff
@@ -95,6 +98,73 @@ autoSaveTest(
...
+		await page.getByLabel('Changeable Default Language').check();
+
+		const updateButton = page.getByRole('button', {
+			name: 'Update',
+		});
+
+		const saveButton = page.getByRole('button', {
+			name: 'Save',
+		});
+
```
Comment (boton) [inline modules/test/playwright/tests/journal-web/journalArticleAutosave.spec.ts:127]: We didn't need it now, I think this is the SF error

**Review by boton — APPROVED** [review]
LGTM. 

I'm sorry, @NemethNorbert. It was my bad. I didn't deploy it or deploy it badly because it works nice and stable ❤️  Nice work on the test!

image

I only left a comment about unused vars SF error.

File: modules/test/playwright/tests/journal-web/journalArticleAutosave.spec.ts (hunk 2 above)
Comment (NemethNorbert) [reply modules/test/playwright/tests/journal-web/journalArticleAutosave.spec.ts:127]: Ohh sorry I missed those lines.

**Comment by NemethNorbert** [comment]
ci:forward

**Pull request closed without merging**